Tests/TuistCacheTests/ContentHashing/ContentHasherTests.swift

```diff
@@ -0,0 +1,76 @@
+import Basic
+import XCTest
+import TuistSupport
+@testable import TuistSupportTesting
+@testable import TuistCache
+
+final class ContentHasherTests: TuistUnitTestCase {
+    private var subject: ContentHasher!
+    private var mockFileHandler: MockFileHandling!
+
+    override func setUp() {
+        super.setUp()
+        mockFileHandler = MockFileHandling()
+        subject = ContentHasher(fileHandler: mockFileHandler)
+    }
+
+    override func tearDown() {
+        subject = nil
+        mockFileHandler = nil
+        super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func test_hashstring_foo_returnsItsMd5() throws {
+        // Given
+        let hash = try subject.hash("foo")
+
+        // Then
+        XCTAssertEqual(hash, "acbd18db4cc2f85cedef654fccc4a4d8")
+    }
+
+    func test_hashstring_bar_returnsItsMd5() throws {
+        // Given
+        let hash = try subject.hash("bar")
+
+        // Then
+        XCTAssertEqual(hash, "37b51d194a7513e45b56f6524f2d51f2")
+    }
+
+    func test_hashstrings_foo_bar_returnsAnotherMd5() throws {
+        // Given
+        let hash = try subject.hash(["foo", "bar"])
+
+        // Then
+        XCTAssertEqual(hash, "3858f62230ac3c915f300c664312c63f")
+    }
+
+    func test_hashFile_HashesTheExpectedFile() throws {
+        // Given
+        let data = Data(repeating: 1, count: 10)
+        mockFileHandler.readFileStub = data
```

Comment:
As I suggested above, I recommend getting rid of the mock that you defined and use the existing mock. You can change this test to be something along the lines of:

```swift
// Given
let tmpDir = try self.temporaryDirectory()
let path = tmpDir + "foo"
let data = Data(repeating: 1, count: 10)
self.fileHandler.write(data, path)
```
Notice how your subject will read real data from disk.

Reply:
`Notice how your subject will read real data from disk.`
🔝 
this is exactly what I'd like to avoid in a unit test. I'm testing the two pieces together in the integration test.

If you don't mind, I'd keep the real mock